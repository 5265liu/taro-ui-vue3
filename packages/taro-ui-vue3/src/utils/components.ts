
export const View = 'taro-view-core'
export const Icon = 'taro-icon-core'
export const Progress = 'taro-progress-core'
export const Text = 'taro-text-core'
export const Button = 'taro-button-core'
export const Checkbox = 'taro-checkbox-core'
export const CheckboxGroup = 'taro-checkbox-group-core'
export const Form = 'taro-form-core'
export const Input = 'taro-input-core'
export const Label = 'taro-label-core'
export const OpenData = 'taro-open-data-core'
export const Picker = 'taro-picker-core'
export const PickerView = 'taro-picker-view-core'
export const PickerViewColumn = 'taro-picker-view-column-core'
export const Radio = 'taro-radio-core'
export const RadioGroup = 'taro-radio-group-core'
export const Slider = 'taro-slider-core'
export const Switch = 'taro-switch-core'
export const Textarea = 'taro-textarea-core'
export const ScrollView = 'taro-scroll-view-core'
export const Swiper = 'taro-swiper-core'
export const SwiperItem = 'taro-swiper-item-core'
export const Image = 'taro-image-core'