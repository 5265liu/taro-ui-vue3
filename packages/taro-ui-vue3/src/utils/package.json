{
  "name": "@taro-ui-vue3/utils",
  "version": "0.0.0",
  "license": "MIT",
  "main": "common.js",
  "peerDependencies": {
    "vue": "^3.0.0",
    "@tarojs/taro": "^3.0.0"
  }
}