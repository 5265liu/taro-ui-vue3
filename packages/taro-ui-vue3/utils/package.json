{
  "name": "@taro-ui-vue3/utils",
  "version": "0.0.0",
  "license": "MIT",
  "main": "common.js",
  "peerDependencies": {
    "@tarojs/taro": "^3.0.0"
  }
}