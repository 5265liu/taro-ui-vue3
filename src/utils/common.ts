import Taro, { SelectorQuery } from '@tarojs/taro'

const { getEnv, ENV_TYPE } = Taro
const ENV = Taro.getEnv()

type ENVS = {
    isWEAPP: boolean,
    isALIPAY: boolean,
    isWEB: boolean,
}

const getEnvs = (): ENVS => {
    const env = getEnv()

    return {
        isWEAPP: env === ENV_TYPE.WEAPP,
        isALIPAY: env === ENV_TYPE.ALIPAY,
        isWEB: env === ENV_TYPE.WEB,
    }    
}

function pxTransform(size: number): string {
    if (!size) return ''

    const designWidth = 750
    const deviceRatio = {
        640: 2.34 / 2,
        750: 1,
        828: 1.81 / 2,
    }

    return `${size / deviceRatio[designWidth]}rpx`
}

function delay(delayTime = 500): Promise<null> {
    return new Promise((resolve) => {
        if ([Taro.ENV_TYPE.WEB, Taro.ENV_TYPE.SWAN].includes(ENV)) {
            setTimeout(() => {
                resolve()
            }, delayTime)
            return
        }
        resolve()
    })
}

function delayQuerySelector(_, selectorStr: string, delayTime = 500): Promise<[]> {
    false && console.log(_)
    const selector: SelectorQuery = Taro.createSelectorQuery()

    return new Promise((resolve) => {
        delay(delayTime).then(() => {
            selector
                .select(selectorStr)
                .boundingClientRect()
                .exec((res: []) => {
                    resolve(res)
                })
        })
    })
}


export { 
    pxTransform,
    getEnvs,
    delayQuerySelector,
}