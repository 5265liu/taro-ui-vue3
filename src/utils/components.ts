import { getEnvs } from './common'
import {
    View as TaroView,
    Icon as TaroIcon,
    Progress as TaroProgress,
    Text as TaroText,
    Button as TaroButton,
    Checkbox as TaroCheckBox,
    CheckboxGroup as TaroCheckBoxGroup,
    Form as TaroForm,
    Input as TaroInput,
    Label as TaroLabel,
    OpenData as TaroOpenData,
    Picker as TaroPicker,
    PickerView as TaroPickerView,
    PickerViewColumn as TaroPickerViewColumn,
    Radio as TaroRadio,
    RadioGroup as TaroRadioGroup,
    Slider as TaroSlider,
    Switch as TaroSwitch,
    Textarea as TaroTextarea,
    ScrollView as TaroScrollView,
    Swiper as TaroSwiper,
    SwiperItem as TaroSwiperItem,
    Image as TaroImage
} from "@tarojs/components";

const isWeb = getEnvs().isWEB

export const View = isWeb ? 'view' : TaroView
export const Icon = isWeb ? 'icon' : TaroIcon
export const Progress = isWeb ? 'progress' : TaroProgress
export const Text = isWeb ? 'text' : TaroText
export const Button = isWeb ? 'button' : TaroButton
export const Checkbox = isWeb ? 'checkbox' : TaroCheckBox
export const CheckboxGroup = isWeb ? 'checkbox-group' : TaroCheckBoxGroup
export const Form = isWeb ? 'form' : TaroForm
export const Input = isWeb ? 'input' : TaroInput
export const Label = isWeb ? 'label' : TaroLabel
export const OpenData = isWeb ? 'open-data' : TaroOpenData
export const Picker = isWeb ? 'picker' : TaroPicker
export const PickerView = isWeb ? 'picker-view' : TaroPickerView
export const PickerViewColumn = isWeb ? 'picker-view-column' : TaroPickerViewColumn
export const Radio = isWeb ? 'radio' : TaroRadio
export const RadioGroup = isWeb ? 'radio-group' : TaroRadioGroup 
export const Slider = isWeb ? 'slider' : TaroSlider
export const Switch = isWeb ? 'switch' : TaroSwitch
export const Textarea = isWeb ? 'textarea' : TaroTextarea
export const ScrollView = isWeb ? 'scroll-view' : TaroScrollView
export const Swiper = isWeb ? 'swiper' : TaroSwiper
export const SwiperItem = isWeb ? 'swiper-item' : TaroSwiperItem
export const Image = isWeb ? 'img' : TaroImage
