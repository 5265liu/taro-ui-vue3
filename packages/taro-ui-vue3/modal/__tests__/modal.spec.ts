import { mount } from '@vue/test-utils'
import AtModal from '../src/index.vue'
import AtModalAction from '../src/modal/action/index.vue'
import AtModalHeader from '../src/modal/header/index.vue'
import AtModalContent from '../src/modal/content/index.vue'
import { sleep } from '@taro-ui-vue3/utils/testHelper'

const components = {
  AtModal,
  AtModalAction,
  AtModalHeader,
  AtModalContent,
}

describe('Modal Snap', () => {
  it('render initial Modal', () => {
    const wrapper = mount({
      components,
      render() {
        return (
          <AtModal>
          <AtModalHeader>标题 < /AtModalHeader>
          < AtModalContent >
          这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
        这里是正文内容，欢迎加入京东凹凸实验室
          < /AtModalContent>
          < AtModalAction >
          <Button>取消 < /Button>
          < Button > 确定 < /Button>
          < /AtModalAction>
          < /AtModal>
        )
  },
    })
expect(wrapper.element).toMatchSnapshot()
  })

it('render opened Modal', () => {
  const wrapper = mount({
    components,
    render() {
      return (
        <AtModal isOpened >
        <AtModalHeader>标题 < /AtModalHeader>
        < AtModalContent >
        这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
      这里是正文内容，欢迎加入京东凹凸实验室
        < /AtModalContent>
        < AtModalAction >
        <Button>取消 < /Button>
        < Button > 确定 < /Button>
        < /AtModalAction>
        < /AtModal>
        )
},
    })
expect(wrapper.element).toMatchSnapshot()
  })

it('render opened Modal -- not header', () => {
  const wrapper = mount({
    components,
    render() {
      return (
        <AtModal isOpened >
        <AtModalContent>
        这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
      这里是正文内容，欢迎加入京东凹凸实验室
        < /AtModalContent>
        < AtModalAction >
        <Button>取消 < /Button>
        < Button > 确定 < /Button>
        < /AtModalAction>
        < /AtModal>
        )
},
    })
expect(wrapper.element).toMatchSnapshot()
  })

it('render opened Modal -- single button', () => {
  const wrapper = mount({
    components,
    render() {
      return (
        <AtModal isOpened >
        <AtModalContent>
        这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
      这里是正文内容，欢迎加入京东凹凸实验室
        < /AtModalContent>
        < AtModalAction >
        <Button>确定 < /Button>
        < /AtModalAction>
        < /AtModal>
        )
},
    })
expect(wrapper.element).toMatchSnapshot()
  })

it('render opened  Modal -- simple', () => {
  const wrapper = mount({
    components,
    render() {
      return (
        <AtModal
            isOpened
            title = "标题"
      cancelText = "取消"
      confirmText = "确认"
      content = "欢迎加入京东凹凸实验室\n\r欢迎加入京东凹凸实验室"
        />
        )
},
    })
expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Modal Behavior ', () => {
  it('Modal onClose & onCancel & onClick', async () => {
    const onCancel = jest.fn()
    const onConfirm = jest.fn()
    const onClose = jest.fn()
    const wrapper = mount({
      components,
      render() {
        return (
          <AtModal
            isOpened
            title = "标题"
        cancelText = "取消"
        confirmText = "确认"
        closeOnClickOverlay
        props = {{
          onClose: onClose,
            onCancel: onCancel,
              onConfirm: onConfirm,
            }
      }
            content="欢迎加入京东凹凸实验室\n\r欢迎加入京东凹凸实验室"
      />
        )
  },
    })

const confirmDom = wrapper.find(
  '.at-modal .at-modal__footer .at-modal__action button:last-child'
)

confirmDom.trigger('tap')
expect(onConfirm).toBeCalled()
await sleep(10)

const cancelDom = wrapper.find(
  '.at-modal .at-modal__footer .at-modal__action button:first-child'
)
cancelDom.trigger('tap')
expect(onCancel).toBeCalled()
await sleep(10)

const overlayDom = wrapper.find('.at-modal .at-modal__overlay')
overlayDom.trigger('tap')
await sleep(10)
expect(onClose).toBeCalled()
  })
})
