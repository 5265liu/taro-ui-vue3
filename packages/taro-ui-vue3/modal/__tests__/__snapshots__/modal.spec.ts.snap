// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snap render initial Modal 1`] = `
<view
  class="at-modal"
>
  <view
    class="at-modal__overlay"
  />
  <view
    class="at-modal__container"
  >
    <view
      class="at-modal__header"
    >
      标题
    </view>
    <scroll-view
      class="at-modal__content"
      scrollY="true"
    >
      这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
    </scroll-view>
    <view
      class="at-modal__footer"
    >
      <view
        class="at-modal__action"
      >
        <button>
          取消
        </button>
        <button>
          确定
        </button>
      </view>
    </view>
  </view>
</view>
`;

exports[`Modal Snap render opened  Modal -- simple 1`] = `
<view
  class="at-modal at-modal--active"
>
  <view
    class="at-modal__overlay"
  />
  <view
    class="at-modal__container"
  >
    <view
      class="at-modal__header"
    >
      <view>
        标题
      </view>
    </view>
    <scroll-view
      class="at-modal__content"
      scrollY="true"
    >
      <view
        class="content-simple"
      >
        <view
          dangerouslySetInnerHTML="[object Object]"
        />
      </view>
    </scroll-view>
    <view
      class="at-modal__footer at-modal__footer--simple"
    >
      <view
        class="at-modal__action"
      >
        <button>
          取消
        </button>
        <button>
          确认
        </button>
      </view>
    </view>
  </view>
</view>
`;

exports[`Modal Snap render opened Modal -- not header 1`] = `
<view
  class="at-modal at-modal--active"
>
  <view
    class="at-modal__overlay"
  />
  <view
    class="at-modal__container"
  >
    <scroll-view
      class="at-modal__content"
      scrollY="true"
    >
      这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
    </scroll-view>
    <view
      class="at-modal__footer"
    >
      <view
        class="at-modal__action"
      >
        <button>
          取消
        </button>
        <button>
          确定
        </button>
      </view>
    </view>
  </view>
</view>
`;

exports[`Modal Snap render opened Modal -- single button 1`] = `
<view
  class="at-modal at-modal--active"
>
  <view
    class="at-modal__overlay"
  />
  <view
    class="at-modal__container"
  >
    <scroll-view
      class="at-modal__content"
      scrollY="true"
    >
      这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
    </scroll-view>
    <view
      class="at-modal__footer"
    >
      <view
        class="at-modal__action"
      >
        <button>
          确定
        </button>
      </view>
    </view>
  </view>
</view>
`;

exports[`Modal Snap render opened Modal 1`] = `
<view
  class="at-modal at-modal--active"
>
  <view
    class="at-modal__overlay"
  />
  <view
    class="at-modal__container"
  >
    <view
      class="at-modal__header"
    >
      标题
    </view>
    <scroll-view
      class="at-modal__content"
      scrollY="true"
    >
      这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室 这里是正文内容，欢迎加入京东凹凸实验室
    </scroll-view>
    <view
      class="at-modal__footer"
    >
      <view
        class="at-modal__action"
      >
        <button>
          取消
        </button>
        <button>
          确定
        </button>
      </view>
    </view>
  </view>
</view>
`;
